### DISCUSSION TITLE
`docker run` auto load added certificates in `/usr/local/share/ca-certificates`

### DISCUSSION THREAD

**Discussion by agreedSkiing** [TabbyML/tabby#1535] [Q&A]
**Please describe the feature you want**
To either auto add certificates in `/usr/local/share/ca-certificates` or have some kind of proxy configuration.

**Additional context**
I'm working behind a proxy inside a windows subsystem for linux 2 in a Ubuntu 20.04.X LTS distro and noticed that the docker image could not reach the model needed from the "basic" (`docker run --entrypoint /opt/tabby/bin/tabby-cpu -it -p 8080:8080 -v $HOME/.tabby:/data tabbyml/tabby serve --model TabbyML/StarCoder-1B`) startup of the image.

The workaround I needed to do to make it work from my system was to mount the company certificates into the pod and execute these commands
```bash
$ docker run -it -p 8080:8080 -v /usr/local/share/ca-certificates:/usr/local/share/ca-certificates -v $HOME/.tabby:/data company.registry.com:5050/tabbyml/tabby
$ update-ca-certificates
$ ./opt/tabby/bin/tabby-cpu serve --model TabbyML/StarCoder-1B
```

P.S. Might it be possible to not use root as the default user when running the tabby?

---
Please reply with a 👍 if you want this feature.

**Comment by wsxiaoys** [comment]
As an intermediate workaround, you might consider customizing the docker image a bit? e.g

```
FROM tabbyml/tabby:0.8.3

USER root
RUN mkdir -p /data
RUN chown 1000 /data

USER 1000

ENTRYPOINT ["/data/start.sh"]
```

Then you could put whatever command inside of `/data/start.sh` to customize your need?

**Comment by agreedSkiing** [comment]
> As an intermediate workaround, you might consider customizing the docker image a bit? e.g
> 
> ```
> FROM tabbyml/tabby:0.8.3
> 
> USER root
> RUN mkdir -p /data
> RUN chown 1000 /data
> 
> USER 1000
> 
> ENTRYPOINT ["/data/start.sh"]
> ```
> 
> Then you could put whatever command inside of `/data/start.sh` to customize your need?

Didn't think about that :+1:. My first solution yesterday was to try to build the image myself... which failed after some hours of trying 😄, after that I noted that the build file never changed the user and lead me to the posted workaround 🙂.
